android/demos/compose/src/main/kotlin/com/uber/rib/compose/root/main/logged_in/LoggedInInteractor.kt

```diff
@@ -38,20 +39,24 @@ class LoggedInInteractor(
 
   override fun didBecomeActive(savedInstanceState: Bundle?) {
     super.didBecomeActive(savedInstanceState)
+
     eventStream.observe()
-      .ofType<LoggedInEvent.LogOutClick>()
-      .autoDispose(this)
-      .subscribe { authStream.accept(AuthInfo(false)) }
+      .onEach {
+        when (it) {
+          is LoggedInEvent.LogOutClick -> authStream.accept(AuthInfo(false))
+        }
+      }
+      .launchIn(coroutineScope)
 
     router.attachOffGame(authInfo)
   }
 
-  override fun onStartGame() {
+  override suspend fun onStartGame() {
```

Comment:
do onStartGame and onGameWon need to be suspending?

Reply:
This was a holdover from a previous experiment with the interactor baseclass. We can remove this now as long as we launch a coroutine in the impl to emit to the sharedflow.